cloud-sql/sql-server/sqlalchemy/connection_test.py

```diff
@@ -0,0 +1,77 @@
+from contextlib import contextmanager
+import logging
+import os
+from typing import Dict
+
+import pyodbc
+import pytest
+
+import main
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.usefixtures("tcp_db_connection")
+def test_tcp_connection(tcp_db_connection):
+    assert tcp_db_connection is not None
+
+
+@pytest.mark.usefixtures('tcp_db_connection')
+def test_get(tcp_db_connection):
+    main.create_tables()
+    context = main.get_index_context()
+    assert isinstance(context, dict)
+    assert len(context.get('recent_votes')) >= 0
+    assert context.get('tab_count') >= 0
+    assert context.get('space_count') >= 0
+
+
+env_map = {
+    'SQLSERVER_USER': 'DB_USER',
+    'SQLSERVER_PASSWORD': 'DB_PASS',
+    'SQLSERVER_DATABASE': 'DB_NAME',
+    'SQLSERVER_HOST': 'DB_HOST',
+    'SQLSERVER_CLOUD_SQL_CONNECTION_NAME': 'CLOUD_SQL_CONNECTION_NAME',
+}
+
+
+@pytest.fixture
+def tcp_db_connection():
+    with mapped_env_variables(env_map):
+        yield from _common_setup()
+
+
+def _common_setup():
+    try:
+        pool = main.init_tcp_connection_engine()
+    except pyodbc.OperationalError as e:
+        logger.warning(
+            'Could not connect to the production database. '
+            'If running tests locally, is the cloud_sql_proxy currently running?'
+        )
+        raise e
+
+    with pool.connect() as conn:
+        conn.execute("SELECT GETDATE()")
```

Comment:
Should this setup function create a table here like it does in the other tests?

Reply:
It certainly could. My thinking was that the GET test confirms the table is created and read from. Since the SQL Server process involves all those heavy ORM objects to create a table (as opposed to a simple `CREATE TABLE` string), this can do its job by successfully executing any given query.